lib/sinon/util/core/walk-object.js

```diff
@@ -1,13 +1,19 @@
 "use strict";
 
+var valueToString = require("@sinonjs/commons").valueToString;
+
 var getPropertyDescriptor = require("./get-property-descriptor");
 var walk = require("./walk");
 
+var reFunctionName = /^function\s*([^\s(]+)/;
+
 function walkObject(predicate, object, filter) {
     var called = false;
+    // IE11 doesn't support fn.name
+    var name = predicate.name || (valueToString(predicate).match(reFunctionName) || [null, ""])[1];
```

Comment:
Should this default to something other than an empty string? I'm open to suggestions.